ovn-tester/tests/service_route.py

```diff
@@ -0,0 +1,79 @@
+from collections import namedtuple
+from ovn_context import Context
+from ovn_workload import Namespace
+from ovn_ext_cmd import ExtCmd
+
+import netaddr
+import ovn_load_balancer as lb
+
+ServiceRouteCfg = namedtuple('ServiceRouteCfg', ['n_lb'])
+
+
+DEFAULT_VIP_SUBNET = netaddr.IPNetwork('90.0.0.0/16')
+DEFAULT_VIP_SUBNET6 = netaddr.IPNetwork('9::/32')
+DEFAULT_VIP_PORT = 80
+DEFAULT_BACKEND_PORT = 8080
+
+
+class ServiceRoute(ExtCmd):
+    def __init__(self, config, central_node, worker_nodes, global_cfg):
+        super(ServiceRoute, self).__init__(
+                config, central_node, worker_nodes)
+        test_config = config.get('service_route', dict())
+        self.config = ServiceRouteCfg(
+            n_lb=test_config.get('n_lb', 16)
+        )
+
+    def get_vip_map_str(self, vip, backend, version):
+        prefix = '[' if version == 6 else ''
+        suffix = ']' if version == 6 else ''
+
+        return {
+            f'{prefix}{vip}{suffix}:{DEFAULT_VIP_PORT}':
+            [f'{prefix}{backend}{suffix}:{DEFAULT_BACKEND_PORT}']
+        }
+
+    def provide_vip(self, vip_subnet, backend, index):
+        vip_net = vip_subnet.next(index)
+        vip_ip = vip_net.ip.__add__(1)
+
+        return self.get_vip_map_str(vip_ip, backend, vip_subnet.version)
+
+    def provide_node_load_balancer(self, name, cluster, node, backend,
+                                   version):
+        load_balancer = lb.OvnLoadBalancer(name, cluster.nbctl)
+        vip = node.ext_rp.ip.ip6 if version == 6 else node.ext_rp.ip.ip4
+        load_balancer.add_vips(self.get_vip_map_str(vip, backend, version))
+        load_balancer.add_to_routers([node.gw_router.name])
+        load_balancer.add_to_switches([node.switch.name])
+
+    def run(self, ovn, global_cfg):
+        ns = Namespace(ovn, 'ns_service_route', global_cfg)
+        with Context(ovn, 'service_route', self.config.n_lb, test=self) as ctx:
+            for i in ctx:
+                ports = ovn.provision_ports(2)
+                ns.add_ports(ports)
+
+                p = ports[1]
+                if p.ip:
+                    ovn.load_balancer.add_vips(
+                        self.provide_vip(DEFAULT_VIP_SUBNET, p.ip, i))
+                if p.ip6:
+                    ovn.load_balancer6.add_vips(
+                        self.provide_vip(DEFAULT_VIP_SUBNET6, p.ip6, i))
+
+                for w in ovn.worker_nodes:
+                    index = i * self.config.n_lb + w.id
+                    if p.ip:
+                        self.provide_node_load_balancer(
+                            f'slb-{index}', ovn, w,
+                            p.ip, 4)
```

Comment:
Both for the node load balancer and cluster load balancer I think it would be useful if we could test with more than a single backend.  Can we add a new configuration, e.g., `n_backends` and add `n_backends` backends instead of a single one?  Maybe a good default value is 4 but if it's configurable we can always change it via config.

Reply:
@dceara ack, I will fix it